src/lao.c

```diff
@@ -7,22 +7,25 @@
 #include "ao/ao.h"
 
 /* -- Library Setup/Teardown -- */
+static int has_initialized = 0;
```

Comment:
I know almost nothing about threads, you're the expert :-) But perhaps we could
ao_append_global_option("hasinitialized", "notyet") and use that, since it resides in the libao that both threads are connected to ?